src/discordas.ts

```diff
@@ -209,3 +224,70 @@ async function run(port: number, fileConfig: DiscordBridgeConfig) {
         process.exit(1);
     }
 }
+
+/**
+ * Logs an error which occured during event processing.
+ *
+ * Depending on the error type different log levels are hardcoded.
+ */
+function logOnEventError(err: Error): void {
+    const errTypes = [];
+    // const warn = [EventInternalError, EventTooOldError, NotReadyError, …];
+    const infoTypes = [];
+    const verboseTypes = [EventUnknownError];
+
+    switch (true) {
+        case instanceofsome(err, errTypes): log.error(err);
+        case instanceofsome(err, infoTypes): log.info(err);
+        case instanceofsome(err, verboseTypes): log.verbose(err);
+        default: log.warn(err);
+    }
+}
+
+/**
+ * Records in which way the request was handled.
+ */
+function recordRequestOutcome(request: Request): void {
+    const eventId = request.getData().eventId;
+    request.getPromise()
+        .then(() =>
+            MetricPeg.get.requestOutcome(eventId, false, "success"),
+        )
+        .catch(EventNotHandledError, (e) =>
+            MetricPeg.get.requestOutcome(eventId, false, "dropped"),
+        )
+        .catch((e) =>
+            MetricPeg.get.requestOutcome(eventId, false, "fail"),
+        )
+    ;
+}
+
+/**
+ * Builds a custom BridgeContext with rooms known to the bridge.
+ */
+async function buildOwnContext(
+    roomId: string,
+    store: DiscordStore,
```

Comment:
if `buildOwnContext` is a private method you don't have to pass over store, right?

Reply:
It's not a method, so I have to pass the store (which is scoped to the `run` function).